### PR TITLE
Implement the onboarding UI

### LINKED ISSUE #3: Implement the Onboarding UI
- [ ] Implement splash screen
- [ ] Implement Onboarding UI 

Image

### PR DESCRIPTION
**Pull request by thanhdo29**
## Related issues
Resolve #3 

## Summary

- Check first time user use app
- Create new components 
- Added new translation keys and new color codes

## How to test
-Test first time use app: 
     If you have used the app, delete the storage of the expo app in the settings section.

### CODE REVIEW

File: components/atoms/TransparentButton.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  title: string
+  onPress?: () => void
+} & ButtonProps
+const TransparentButton = (props: Props): React.ReactElement => {
+  const { fonts } = useAppFonts()
+  const colors = getColors(useColorScheme())
+  return (
+    <Button
+      onPress={props.onPress}
+      bg="$colorTransparent"
```
Comment (KMTus) [inline components/atoms/TransparentButton.tsx:19]: You should use the available color prepared, to achieve uniformity in color

File: components/molecules/OnboardingItem.tsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+const OnboardingItem = (props: Props): React.ReactElement => {
+  const colors = getColors(useColorScheme())
+  const { fonts } = useAppFonts()
+  return (
+    <View width={width}>
+      <ImageBackground
+        source={props.imgBackground}
+        style={styles.background}
+      >
+        <View style={styles.overlay} backgroundColor={'rgba(0, 0, 0, 0.3)'} />
```
Comment (KMTus) [inline components/molecules/OnboardingItem.tsx:24]: color

File: components/atoms/TransparentButton.tsx (hunk 1 above)
Comment (thanhdo29) [reply components/atoms/TransparentButton.tsx:19]: It's transparent so I had to use it like that

**Comment by KMTus** [comment]
image
image
good job

**Review by KMTus — APPROVED** [review]

**Review by NguyenVanViet2004 — COMMENTED** [review]
You should hard code the color white for `Get stater` button here

Simulator Screenshot - iPhone 15 Pro - 2024-09-24 at 19 28 06

File: components/atoms/PrimaryButton.tsx
Comment (NguyenVanViet2004) [inline components/atoms/PrimaryButton.tsx:1]: Use this code block, this code block is cleaner than your code

```jsx
import { useColorScheme } from 'react-native'
import { Button, type ButtonProps, Text } from 'tamagui'

import getColors from '~/constants/Colors'
import { RADIUS_BUTTON } from '~/constants/Constants'
import { useAppFonts } from '~/hooks/useAppFonts'

type Props = {
  title: string
} & ButtonProps
const PrimaryButton = (props: Props): React.ReactElement => {
  const { fonts } = useAppFonts()
  const colors = getColors(useColorScheme())
  return (
    <Button
      borderRadius={RADIUS_BUTTON}
      alignItems="center"
      onPress={props.onPress}>
      <Text
        color={colors.blueSapphire}
        fontFamily={fonts.JetBrainsMonoBold}
        fontSize={16}>
        {props.title}
      </Text>
    </Button>
  )
}

export default PrimaryButton

```

File: components/atoms/TransparentButton.tsx
Comment (NguyenVanViet2004) [inline components/atoms/TransparentButton.tsx:1]:
```jsx
import React from 'react'
import { useColorScheme } from 'react-native'
import { Button, type ButtonProps, Text } from 'tamagui'

import getColors from '~/constants/Colors'
import { RADIUS_BUTTON } from '~/constants/Constants'
import { useAppFonts } from '~/hooks/useAppFonts'

type Props = {
  title: string
} & ButtonProps
const TransparentButton = (props: Props): React.ReactElement => {
  const { fonts } = useAppFonts()
  const colors = getColors(useColorScheme())
  return (
    <Button
      onPress={props.onPress}
      bg="$colorTransparent"
      borderRadius={RADIUS_BUTTON}
      borderColor={colors.white}
      borderWidth={1}>
      <Text
        fontFamily={fonts.JetBrainsMonoBold}
        color={colors.white}
        fontSize={16}>
        {props.title}
      </Text>
    </Button>
  )
}

export default TransparentButton

```

File: components/molecules/OnboardingItem.tsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+const OnboardingItem = (props: Props): React.ReactElement => {
+  const colors = getColors(useColorScheme())
+  const { fonts } = useAppFonts()
+  return (
+    <View width={width}>
+      <ImageBackground
+        source={props.imgBackground}
+        style={styles.background}
+      >
```
Comment (NguyenVanViet2004) [inline components/molecules/OnboardingItem.tsx:23]:
```suggestion
      style={styles.background}>
```

File: components/templates/OnboardingTemplate.tsx [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+  const scrollX = useRef(new Animated.Value(0)).current
+  const slideRef = useRef<FlatList<number>>(null)
+  const colors = getColors(useColorScheme())
+  const { width } = useWindowDimensions()
+  const { t } = useTranslation()
+
+  const viewableItemsChanged = useRef(({ viewableItems }:
+  { viewableItems: ViewToken[] }) => {
+    setCurrentIndex(viewableItems[0].index ?? 0)
+    if (viewableItems[0]?.index === 1 ||
```
Comment (NguyenVanViet2004) [inline components/templates/OnboardingTemplate.tsx:24]:
```suggestion
    if (viewableItems[0]?.index >= 1 && viewableItems[0]?.index < 4 )
```

**Review by NguyenVanViet2004 — CHANGES_REQUESTED** [review]

**Review by NguyenVanViet2004 — APPROVED** [review]
Ảnh màn hình 2024-09-24 lúc 21 25 45

**Review by KMTus — APPROVED** [review]

**Pull request merged**